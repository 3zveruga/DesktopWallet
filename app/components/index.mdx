---
name: Home
route: /
---

# UI Components

Use the sidebar menu to browse the styleguide
