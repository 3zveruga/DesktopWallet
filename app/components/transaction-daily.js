// @flow

import React, { Fragment } from 'react';
import styled from 'styled-components';

import { TransactionItemComponent, type Transaction } from './transaction-item';
import { TextComponent } from './text';
import { Divider } from './divider';

const Wrapper = styled.div`
  margin-top: 20px;
`;

const TransactionsWrapper = styled.div`
  border-radius: ${props => props.theme.boxBorderRadius};
  overflow: hidden;
  background-color: ${props => props.theme.colors.cardBackgroundColor};
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  margin-bottom: 20px;
`;

const Day = styled(TextComponent)`
  text-transform: uppercase;
  color: ${props => props.theme.colors.transactionsDate};
  font-size: ${props => `${props.theme.fontSize.regular * 0.9}em`};
  font-weight: ${props => props.theme.fontWeight.bold};
  margin-bottom: 5px;
`;

type Props = {
  transactionsDate: string,
  transactions: Transaction[],
  zecPrice: number,
};

export const TransactionDailyComponent = ({
  transactionsDate,
  transactions,
  zecPrice,
}: Props) => (
  <Wrapper data-testid='TransactionsDaily'>
    <Day value={transactionsDate} />
    <TransactionsWrapper>
      {transactions.map(
        ({
          date, type, address, amount, transactionId,
        }, idx) => (
          <Fragment key={`${address}-${type}-${amount}-${date}`}>
            <TransactionItemComponent
              transactionId={transactionId}
              type={type}
              date={date}
              address={address || ''}
              amount={amount}
              zecPrice={zecPrice}
            />
            {idx < transactions.length - 1 && <Divider />}
          </Fragment>
        ),
      )}
    </TransactionsWrapper>
  </Wrapper>
);
