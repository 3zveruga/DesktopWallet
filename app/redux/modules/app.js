// @flow
import type { Action } from '../../types/redux';

// Actions
export const SHOW_ERROR_MODAL = 'SHOW_ERROR_MODAL';
export const HIDE_ERROR_MODAL = 'HIDE_ERROR_MODAL';

export const showErrorModal = ({ error }: { error: string }) => ({
  type: SHOW_ERROR_MODAL,
  payload: {
    error,
  },
});

export const closeErrorModal = () => ({
  type: HIDE_ERROR_MODAL,
  payload: {},
});

export type State = {
  isErrorModalVisible: boolean,
  error: string | null,
};

const initialState: State = {
  isErrorModalVisible: false,
  error: null,
};

// eslint-disable-next-line
export default (state: State = initialState, action: Action) => {
  switch (action.type) {
    case SHOW_ERROR_MODAL:
      return { isErrorModalVisible: true, error: action.payload.error };
    case HIDE_ERROR_MODAL:
      return { isErrorModalVisible: false, error: null };
    default:
      return state;
  }
};
