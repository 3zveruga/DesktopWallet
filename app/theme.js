// @flow

import React, { Fragment, type Node } from 'react';
import theme from 'styled-theming';
import { ThemeProvider, createGlobalStyle } from 'styled-components';
// $FlowFixMe
import { normalize } from 'polished'; // eslint-disable-line

import { DARK } from './constants/themes';

const darkOne = '#F4B728';
const blackTwo = '#171717';
const lightOne = '#ffffff';
const brandOne = '#000';
const brandThree = '#5d5d65';
const buttonBorderColor = '#3e3c42';
const activeItem = '#F4B728';
const text = '#FFF';
const cardBackgroundColor = '#000';
const sidebarLogoGradientBegin = '#F4B728';
const sidebarLogoGradientEnd = '#FFE240';
const sidebarHoveredItem = '#1C1C1C';
const sidebarHoveredItemLabel = '#8e8e96';
const background = '#212124';
const transactionSent = '#FF6C6C';
const transactionReceived = '#6AEAC0';
const transactionsDate = '#777777';
const transactionsItemHovered = '#222222';
const selectButtonShadow = 'rgba(238,201,76,0.65)';
const statusPillLabel = '#727272';
const transactionsDetailsLabel = transactionsDate;

const appTheme: AppTheme = {
  mode: DARK,
  fontFamily: 'Roboto',
  fontWeight: {
    bold: 700,
    default: 400,
    light: 300,
  },
  fontSize: {
    large: 1.25,
    medium: 1.125,
    regular: 0.84375,
    small: 0.667,
  },
  colors: {
    primary: theme('mode', {
      light: lightOne,
      dark: darkOne,
    }),
    secondary: theme('mode', {
      light: darkOne,
      dark: lightOne,
    }),
    sidebarBg: theme('mode', {
      light: brandOne,
      dark: brandOne,
    }),
    sidebarItem: theme('mode', {
      light: brandThree,
      dark: brandThree,
    }),
    sidebarItemActive: theme('mode', {
      light: activeItem,
      dark: activeItem,
    }),
    sidebarHoveredItem: theme('mode', {
      light: sidebarHoveredItem,
      dark: sidebarHoveredItem,
    }),
    sidebarHoveredItemLabel: theme('mode', {
      light: sidebarHoveredItemLabel,
      dark: sidebarHoveredItemLabel,
    }),
    cardBackgroundColor: theme('mode', {
      light: cardBackgroundColor,
      dark: cardBackgroundColor,
    }),
    text: theme('mode', {
      light: '#000',
      dark: text,
    }),
    activeItem: theme('mode', {
      light: activeItem,
      dark: activeItem,
    }),
    inactiveItem: theme('mode', {
      light: brandThree,
      dark: brandThree,
    }),
    sidebarLogoGradientBegin: theme('mode', {
      light: sidebarLogoGradientBegin,
      dark: sidebarLogoGradientBegin,
    }),
    sidebarLogoGradientEnd: theme('mode', {
      light: sidebarLogoGradientEnd,
      dark: sidebarLogoGradientEnd,
    }),
    background: theme('mode', {
      light: '#FFF',
      dark: background,
    }),
    transactionSent: theme('mode', {
      light: transactionSent,
      dark: transactionSent,
    }),
    transactionReceived: theme('mode', {
      light: transactionReceived,
      dark: transactionReceived,
    }),
    transactionsDate: theme('mode', {
      light: transactionsDate,
      dark: transactionsDate,
    }),
    transactionsItemHovered: theme('mode', {
      light: transactionsItemHovered,
      dark: transactionsItemHovered,
    }),
    inputBackground: theme('mode', {
      light: brandOne,
      dark: brandOne,
    }),
    selectButtonShadow: theme('mode', {
      light: selectButtonShadow,
      dark: selectButtonShadow,
    }),
    transactionsDetailsLabel: theme('mode', {
      light: transactionsDetailsLabel,
      dark: transactionsDetailsLabel,
    }),
    statusPillLabel: theme('mode', {
      light: statusPillLabel,
      dark: statusPillLabel,
    }),
    modalItemLabel: theme('mode', {
      light: transactionsDate,
      dark: transactionsDate,
    }),
    blackTwo: theme('mode', {
      light: blackTwo,
      dark: blackTwo,
    }),
    buttonBorderColor: theme('mode', {
      light: buttonBorderColor,
      dark: buttonBorderColor,
    }),
  },
  sidebarWidth: '180px',
  headerHeight: '60px',
  layoutPaddingLeft: '35px',
  layoutPaddingRight: '35px',
  layoutContentPaddingTop: '20px',
  boxBorderRadius: '3px',
  transitionEase: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)',
};

export const GlobalStyle = createGlobalStyle`
  ${normalize()}

  * {
    box-sizing: border-box;
  }
`;

export const DoczWrapper = ({ children }: { children: () => Node<*> }) => (
  <ThemeProvider theme={appTheme}>
    <Fragment>
      <GlobalStyle />
      {children()}
    </Fragment>
  </ThemeProvider>
);

// eslint-disable-next-line
export default appTheme;
