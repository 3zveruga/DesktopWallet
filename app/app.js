// @flow

import React, { Component, Fragment } from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'styled-components';

import { configureStore, history } from './redux/create';
import { Router } from './router/container';
import theme, { GlobalStyle } from './theme';
import electronStore from '../config/electron-store';
import { DARK } from './constants/themes';

import 'rc-tooltip/assets/bootstrap.css';

const store = configureStore({});

type Props = {};
type State = {
  themeMode: string,
};

export class App extends Component<Props, State> {
  state = {
    themeMode: electronStore.get('THEME_MODE') || DARK,
  };

  componentDidMount() {
    if (!electronStore.has('THEME_MODE')) {
      electronStore.set('THEME_MODE', DARK);
    }

    electronStore.onDidChange('THEME_MODE', newValue => this.setState({ themeMode: newValue }));
  }

  render() {
    const { themeMode } = this.state;

    return (
      <ThemeProvider theme={{ ...theme, mode: themeMode }}>
        <Fragment>
          <GlobalStyle />
          <Provider store={store}>
            <ConnectedRouter history={history}>
              <Router />
            </ConnectedRouter>
          </Provider>
        </Fragment>
      </ThemeProvider>
    );
  }
}
