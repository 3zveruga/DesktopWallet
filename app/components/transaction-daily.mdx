---
name: Transaction Daily
---

import { Playground, PropsTable } from 'docz'

import { TransactionDailyComponent } from './transaction-daily.js'
import { DoczWrapper } from '../theme.js'

# Transaction Item

## Properties

<PropsTable of={TransactionDailyComponent} />

## Basic Usage

<Playground>
  <DoczWrapper>
    {() => (
      <TransactionDailyComponent
        transactionsDate={new Date().toISOString()}
        transactions={[
          {
            type: 'received',
            address: '123456789123456789123456789123456789',
            amount: 1.7891,
            date: new Date().toISOString(),
          },
          {
            type: 'sent',
            address: '123456789123456789123456789123456789',
            amount: 0.8458,
            date: new Date().toISOString(),
          },
        ]}
      />
    )}
  </DoczWrapper>
</Playground>
