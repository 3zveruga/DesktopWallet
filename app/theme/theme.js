// @flow

import theme from 'styled-theming';

import { DARK, LIGHT } from '../constants/themes';
import { typography } from './typography';
import { DARK_COLORS, LIGHT_COLORS } from './colors';

export const appTheme: AppTheme = {
  // General
  mode: DARK,

  // Typography
  ...typography,

  // Spacing
  sidebarWidth: '180px',
  headerHeight: '60px',
  layoutPaddingLeft: '35px',
  layoutPaddingRight: '35px',
  layoutContentPaddingTop: '20px',
  boxBorderRadius: '3px',

  // Misc
  transitionEase: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)',

  // Colors
  colors: {
    primary: theme('mode', {
      [LIGHT]: LIGHT_COLORS.darkOne,
      [DARK]: DARK_COLORS.darkOne,
    }),
    secondary: theme('mode', {
      [LIGHT]: LIGHT_COLORS.darkOne,
      [DARK]: DARK_COLORS.darkOne,
    }),
    divider: theme('mode', {
      [LIGHT]: LIGHT_COLORS.divider,
      [DARK]: DARK_COLORS.divider,
    }),
    sidebarBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarBg,
      [DARK]: DARK_COLORS.sidebarBg,
    }),
    sidebarItem: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarItem,
      [DARK]: DARK_COLORS.sidebarItem,
    }),
    sidebarItemActive: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarItemActive,
      [DARK]: DARK_COLORS.sidebarItemActive,
    }),
    sidebarItemHovered: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarItemHovered,
      [DARK]: DARK_COLORS.sidebarItemHovered,
    }),
    sidebarHoveredItemLabel: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarHoveredItemLabel,
      [DARK]: DARK_COLORS.sidebarHoveredItemLabel,
    }),
    cardBackgroundColor: theme('mode', {
      [LIGHT]: LIGHT_COLORS.cardBackgroundColor,
      [DARK]: DARK_COLORS.cardBackgroundColor,
    }),
    text: theme('mode', {
      [LIGHT]: LIGHT_COLORS.text,
      [DARK]: DARK_COLORS.text,
    }),
    activeItem: theme('mode', {
      [LIGHT]: LIGHT_COLORS.activeItem,
      [DARK]: DARK_COLORS.activeItem,
    }),
    inactiveItem: theme('mode', {
      [LIGHT]: LIGHT_COLORS.brandThree,
      [DARK]: DARK_COLORS.brandThree,
    }),
    sidebarLogoGradientBegin: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarLogoGradientBegin,
      [DARK]: DARK_COLORS.sidebarLogoGradientBegin,
    }),
    sidebarLogoGradientEnd: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarLogoGradientEnd,
      [DARK]: DARK_COLORS.sidebarLogoGradientEnd,
    }),
    background: theme('mode', {
      [LIGHT]: LIGHT_COLORS.background,
      [DARK]: DARK_COLORS.background,
    }),
    transactionSent: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionSent,
      [DARK]: DARK_COLORS.transactionSent,
    }),
    transactionReceived: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionReceived,
      [DARK]: DARK_COLORS.transactionReceived,
    }),
    transactionsDate: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionsDate,
      [DARK]: DARK_COLORS.transactionsDate,
    }),
    transactionsItemHovered: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionsItemHovered,
      [DARK]: DARK_COLORS.transactionsItemHovered,
    }),
    inputBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.inputBg,
      [DARK]: DARK_COLORS.inputBg,
    }),
    selectButtonShadow: theme('mode', {
      [LIGHT]: LIGHT_COLORS.selectButtonShadow,
      [DARK]: DARK_COLORS.selectButtonShadow,
    }),
    transactionDetailsLabel: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionDetailsLabel,
      [DARK]: DARK_COLORS.transactionDetailsLabel,
    }),
    statusPillLabel: theme('mode', {
      [LIGHT]: LIGHT_COLORS.statusPillLabel,
      [DARK]: DARK_COLORS.statusPillLabel,
    }),
    modalItemLabel: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionsDate,
      [DARK]: DARK_COLORS.transactionsDate,
    }),
    blackTwo: theme('mode', {
      [LIGHT]: LIGHT_COLORS.blackTwo,
      [DARK]: DARK_COLORS.blackTwo,
    }),
    buttonBorderColor: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonBorderColor,
      [DARK]: DARK_COLORS.buttonBorderColor,
    }),
    headerTitle: theme('mode', {
      [LIGHT]: LIGHT_COLORS.headerTitle,
      [DARK]: DARK_COLORS.headerTitle,
    }),
    statusPillBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.statusPillBg,
      [DARK]: DARK_COLORS.statusPillBg,
    }),
    walletSummaryBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletSummaryBg,
      [DARK]: DARK_COLORS.walletSummaryBg,
    }),
    walletSummaryBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletSummaryBorder,
      [DARK]: DARK_COLORS.walletSummaryBorder,
    }),
    consoleBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.consoleBg,
      [DARK]: DARK_COLORS.consoleBg,
    }),
    sidebarBorderRight: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarBorderRight,
      [DARK]: DARK_COLORS.sidebarBorderRight,
    }),
    qrCodeWrapperBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.qrCodeWrapperBg,
      [DARK]: DARK_COLORS.qrCodeWrapperBg,
    }),
    statusPillBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.statusPillBorder,
      [DARK]: DARK_COLORS.statusPillBorder,
    }),
    sidebarActiveItemBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarActiveItemBorder,
      [DARK]: DARK_COLORS.sidebarActiveItemBorder,
    }),
    transactionItemBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionItemBg,
      [DARK]: DARK_COLORS.transactionItemBg,
    }),
    transactionItemHoverBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionItemHoverBg,
      [DARK]: DARK_COLORS.transactionItemHoverBg,
    }),
    transactionItemBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionItemBorder,
      [DARK]: DARK_COLORS.transactionItemBorder,
    }),
    transactionItemAddress: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionItemAddress,
      [DARK]: DARK_COLORS.transactionItemAddress,
    }),
    transactionItemAddressHover: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionItemAddressHover,
      [DARK]: DARK_COLORS.transactionItemAddressHover,
    }),
    transactionDetailsShadow: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionDetailsShadow,
      [DARK]: DARK_COLORS.transactionDetailsShadow,
    }),
    transactionDetailsBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionDetailsBg,
      [DARK]: DARK_COLORS.transactionDetailsBg,
    }),
    transactionDetailsRowHover: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionDetailsRowHover,
      [DARK]: DARK_COLORS.transactionDetailsRowHover,
    }),
    transactionDetailsDivider: theme('mode', {
      [LIGHT]: LIGHT_COLORS.transactionDetailsDivider,
      [DARK]: DARK_COLORS.transactionDetailsDivider,
    }),
    inputBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.inputBorder,
      [DARK]: DARK_COLORS.inputBorder,
    }),
    sidebarItemHoveredBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sidebarItemHoveredBg,
      [DARK]: DARK_COLORS.sidebarItemHoveredBg,
    }),
    consoleBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.consoleBorder,
      [DARK]: DARK_COLORS.consoleBorder,
    }),
    sendCardBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sendCardBg,
      [DARK]: DARK_COLORS.sendCardBg,
    }),
    sendCardBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sendCardBorder,
      [DARK]: DARK_COLORS.sendCardBorder,
    }),
    walletAddressBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletAddressBg,
      [DARK]: DARK_COLORS.walletAddressBg,
    }),
    walletAddressBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletAddressBorder,
      [DARK]: DARK_COLORS.walletAddressBorder,
    }),
    walletAddressInput: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletAddressInput,
      [DARK]: DARK_COLORS.walletAddressInput,
    }),
    walletAddressInputHovered: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletAddressInputHovered,
      [DARK]: DARK_COLORS.walletAddressInputHovered,
    }),
    dropdownBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.dropdownBg,
      [DARK]: DARK_COLORS.dropdownBg,
    }),
    dropdownHoveredBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.dropdownHoveredBg,
      [DARK]: DARK_COLORS.dropdownHoveredBg,
    }),
    dropdownBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.dropdownBorder,
      [DARK]: DARK_COLORS.dropdownBorder,
    }),
    settingsCardBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.settingsCardBg,
      [DARK]: DARK_COLORS.settingsCardBg,
    }),
    settingsLearnMore: theme('mode', {
      [LIGHT]: LIGHT_COLORS.settingsLearnMore,
      [DARK]: DARK_COLORS.settingsLearnMore,
    }),
    settingsLearnMoreHovered: theme('mode', {
      [LIGHT]: LIGHT_COLORS.settingsLearnMoreHovered,
      [DARK]: DARK_COLORS.settingsLearnMoreHovered,
    }),
    buttonPrimaryText: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonPrimaryText,
      [DARK]: DARK_COLORS.buttonPrimaryText,
    }),
    buttonSecondaryText: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonSecondaryText,
      [DARK]: DARK_COLORS.buttonSecondaryText,
    }),
    buttonPrimaryBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonPrimaryBg,
      [DARK]: DARK_COLORS.buttonPrimaryBg,
    }),
    buttonSecondaryBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonSecondaryBg,
      [DARK]: DARK_COLORS.buttonSecondaryBg,
    }),
    buttonPrimaryDisabledBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonPrimaryDisabledBg,
      [DARK]: DARK_COLORS.buttonPrimaryDisabledBg,
    }),
    buttonSecondaryDisabledBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonSecondaryDisabledBg,
      [DARK]: DARK_COLORS.buttonSecondaryDisabledBg,
    }),
    buttonSecondaryBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonSecondaryBorder,
      [DARK]: DARK_COLORS.buttonSecondaryBorder,
    }),
    buttonSecondaryHoveredBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.buttonSecondaryHoveredBg,
      [DARK]: DARK_COLORS.buttonSecondaryHoveredBg,
    }),
    dropdownIconBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.dropdownIconBorder,
      [DARK]: DARK_COLORS.dropdownIconBorder,
    }),
    dropdownOpenedIconBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.dropdownOpenedIconBorder,
      [DARK]: DARK_COLORS.dropdownOpenedIconBorder,
    }),
    inputBorderActive: theme('mode', {
      [LIGHT]: LIGHT_COLORS.inputBorderActive,
      [DARK]: DARK_COLORS.inputBorderActive,
    }),
    walletAddressTooltipBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletAddressTooltipBg,
      [DARK]: DARK_COLORS.walletAddressTooltipBg,
    }),
    qrCodeWrapperBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.qrCodeWrapperBorder,
      [DARK]: DARK_COLORS.qrCodeWrapperBorder,
    }),
    walletAddressTooltip: theme('mode', {
      [LIGHT]: LIGHT_COLORS.walletAddressTooltip,
      [DARK]: DARK_COLORS.walletAddressTooltip,
    }),
    loadingScreenBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.loadingScreenBg,
      [DARK]: DARK_COLORS.loadingScreenBg,
    }),
    loadingScreenText: theme('mode', {
      [LIGHT]: LIGHT_COLORS.loadingScreenText,
      [DARK]: DARK_COLORS.loadingScreenText,
    }),
    sendAdditionalOptionsBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sendAdditionalOptionsBg,
      [DARK]: DARK_COLORS.sendAdditionalOptionsBg,
    }),
    sendAdditionalOptionsBorder: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sendAdditionalOptionsBorder,
      [DARK]: DARK_COLORS.sendAdditionalOptionsBorder,
    }),
    sendAdditionalInputBg: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sendAdditionalInputBg,
      [DARK]: DARK_COLORS.sendAdditionalInputBg,
    }),
    sendAdditionalInputText: theme('mode', {
      [LIGHT]: LIGHT_COLORS.sendAdditionalInputText,
      [DARK]: DARK_COLORS.sendAdditionalInputText,
    }),
  },
};
