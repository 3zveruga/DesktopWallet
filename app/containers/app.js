// @flow
import { connect } from 'react-redux';
import { closeErrorModal } from '../redux/modules/app';

import { LayoutComponent } from '../components/layout';

import type { Dispatch } from '../types/redux';
import type { AppState } from '../types/app-state';

const mapStateToProps = ({ app }: AppState) => ({
  isErrorModalVisible: app.isErrorModalVisible,
  error: app.error,
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  closeErrorModal: () => dispatch(closeErrorModal()),
});

// $FlowFixMe
export const AppContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(LayoutComponent);
