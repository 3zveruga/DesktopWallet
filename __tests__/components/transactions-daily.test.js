// @flow

import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import 'jest-dom/extend-expect';

import { TransactionDailyComponent } from '../../app/components/transaction-daily';
import { appTheme } from '../../app/theme';

afterEach(cleanup);

describe('<TransactionDailyComponent />', () => {
  describe('render()', () => {
    test('should render user daily transactions', () => {
      const { container } = render(
        <ThemeProvider theme={appTheme}>
          <TransactionDailyComponent
            transactionsDate='2019-02-20T19:31:57.117Z'
            zecPrice={1.345}
            transactions={[
              {
                type: 'receive',
                transactionId: 's0a8das098fgh2348a',
                address: '123456789123456789123456789123456789',
                amount: 1.7891,
                zecPrice: 1.345,
                date: '2019-02-20T19:31:57.117Z',
                theme: appTheme,
                fees: 0.001,
              },
              {
                type: 'send',
                transactionId: '0asd908fgj90f01',
                address: '123456789123456789123456789123456789',
                amount: 0.8458,
                zecPrice: 1.344,
                date: '2019-02-20T19:31:57.117Z',
                theme: appTheme,
                fees: 0.001,
              },
            ]}
          />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
