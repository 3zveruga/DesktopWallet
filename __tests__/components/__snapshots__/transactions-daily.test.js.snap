// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionDailyComponent /> render() should render user daily transactions 1`] = `
<div>
  <div
    class="sc-kpOJdX inMzwo"
    data-testid="TransactionsDaily"
  >
    <p
      class="sc-ckVGcZ kFpbuU sc-htpNat pRhzD"
    >
      2019-02-20T19:31:57.117Z
    </p>
    <div
      class="sc-dxgOiQ dHRuWE"
    >
      <div
        class="sc-jzJRlG bgSReS sc-bdVaJa bWRfen"
        id="transaction-item-s0a8das098fgh2348a"
      >
        <div
          class="sc-bdVaJa eFCRqo"
        >
          <div
            class="sc-bdVaJa eFCRqo"
          >
            <img
              alt="Transaction Type Icon"
              class="sc-cSHVUG fQARKX"
              src="[object Object]"
            />
            <div
              class="sc-kGXeez iHFhdz sc-bwzfXH iiePXZ"
              width=""
            >
              <p
                class="sc-kAzzGY edxhBx sc-htpNat ccIOzB"
              >
                receive
              </p>
              <p
                class="sc-kgoBCf bfmmtg sc-htpNat pRhzD"
              >
                16:31 PM
              </p>
            </div>
          </div>
          <p
            class="sc-chPdSV bQobpM sc-htpNat pRhzD"
          >
            12345678912345678912...9123456789
          </p>
        </div>
        <div
          class="sc-bwzfXH jExuEg"
          width=""
        >
          <p
            class="sc-htpNat dztgbW"
            color="#6AEAC0"
          >
            +ZEC 1.789
          </p>
          <p
            class="sc-htpNat dhSLjI"
            color="#5d5d65"
          >
            +USD $2.406
          </p>
        </div>
      </div>
      <div
        class="sc-jzJRlG bgSReS sc-bdVaJa bWRfen"
        id="transaction-item-0asd908fgj90f01"
      >
        <div
          class="sc-bdVaJa eFCRqo"
        >
          <div
            class="sc-bdVaJa eFCRqo"
          >
            <img
              alt="Transaction Type Icon"
              class="sc-cSHVUG fQARKX"
              src="[object Object]"
            />
            <div
              class="sc-kGXeez iHFhdz sc-bwzfXH iiePXZ"
              width=""
            >
              <p
                class="sc-kAzzGY gABhMI sc-htpNat ccIOzB"
              >
                send
              </p>
              <p
                class="sc-kgoBCf bfmmtg sc-htpNat pRhzD"
              >
                16:31 PM
              </p>
            </div>
          </div>
          <p
            class="sc-chPdSV bQobpM sc-htpNat pRhzD"
          >
            12345678912345678912...9123456789
          </p>
        </div>
        <div
          class="sc-bwzfXH jExuEg"
          width=""
        >
          <p
            class="sc-htpNat jAbxuE"
            color="#FF6C6C"
          >
            -ZEC 0.846
          </p>
          <p
            class="sc-htpNat dhSLjI"
            color="#5d5d65"
          >
            -USD $1.138
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
